src/App/App.js

```diff
@@ -1,35 +1,47 @@
 import './App.scss';
 import Nav from '../Nav/Nav';
 import React, { Component } from 'react';
-import movieData from '../movieData';
+//import movieData from '../movieData';
 import MovieContainer from '../MovieContainer/MovieContainer';
 import Movie from '../Movie/Movie';
+import { callApi, callSingleApi } from '../callApis'
 
 class App extends Component {
   constructor() {
     super()
     this.state = {
-      movies: movieData.movies,
-      movieSelected: {} 
+      movies: [],
+      movieSelected: {},
+      error: null
     }
   }
+  //integration test, tests the methods
+  handleClick = (id) => {
+    //fetch her for single movie
+    // .then(data => this.setState({movieSelected: data}))
+    callSingleApi(id)
+      .then(data => this.setState({ movieSelected: data.movie }))
+      .catch(error => this.setState({ error }))
+  }
 
-  handleClick = (index) => {
-    this.setState({ movieSelected: this.state.movies[index] })
+  returnHome = () => {
+    this.setState({ movieSelected: {} })
   }
 
-  handleSubmit = () => {
-    this.setState( { movies: movieData.movies, movieSelected: {} })
+  componentDidMount() {
+    callApi()
+      .then(data => this.setState({ movies: data.movies }))
+      .catch(error => this.setState({ error }))
   }
 
   render() {
     return (
       <div className="App">
-        <Nav handleSubmit={ this.handleSubmit }/>
+        <Nav returnHome={ this.returnHome } />
         { Object.keys(this.state.movieSelected).length ? <Movie movieSelected={ this.state.movieSelected } /> : <MovieContainer
```

Comment:
Wow...`Object.keys(this.state.movieSelected).length`

Probably would have been easier if you just `setState({ movieSelected: null })` in `returnHome`